### ISSUE TITLE
Move newtype pattern section for `Component` to the Book

### ISSUE THREAD

**Issue by Nilirad** [bevyengine/bevy-website#375]
## Issue

The following PR:

- bevyengine/bevy#4767,

introduces a paragraph in the API docs for `Component` about using the newtype pattern to implement the trait for foreign types. However, both @alice-i-cecile and myself, believe that this type of information is more suitable to the Bevy book.

## Solution

The following steps should be completed in order.

- [ ] Add paragraph and example to the Bevy book.
- [ ] Publish the version of the Bevy book containing the aforementioned change.
- [ ] Replace the paragraph and example in the API docs with a short sentence with a link to the right section of the Bevy book.

**Comment by TrialDragon** [comment]
Where in the book would this paragraph go? Feels a bit more in-depth than the current ECS page in Getting Started goes imo; has more of a Component page in proper book feeling, but there isn't one at this time.

**Comment by alice-i-cecile** [comment]
Yeah, I'm not sure of the best place for it currently. The single most helpful thing to do would be to tackle the work laid out in #623: I will drop all other work to review (and wrangle reviewrs) on progress there.

Once we have a solid base to work off of, changes like this can be tackled more appropriately.

**Comment by TrialDragon** [comment]
Alright. That was already my next planned issue to tackle once I polish off the ones I started.
